app/src/main/java/com/example/solar_alarm/SetAlarmActivity.java

```diff
@@ -0,0 +1,73 @@
+package com.example.solar_alarm;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.app.AlarmManager;
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Build;
+import android.os.Bundle;
+import android.os.SystemClock;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TimePicker;
+
+import java.text.DateFormat;
+import java.util.Calendar;
+
+
+public class SetAlarmActivity extends AppCompatActivity {
+    private TimePicker timePicker;
+    private int hour;
+    private int min;
+    private EditText alarmName;
+    private Button setAlarm;
+
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
```

Comment:
This should be an edit method. The button is also creating an infinite look because each click brings it back to the create screen.